JOANNE/main.yaml

```diff
@@ -4,10 +4,14 @@ plugins:
 
 sources:
   level3:
-    driver: opendap
```

Comment:
@d70-t This seems a great approach. What's the `simplecache::` syntax?

Reply:
`fsspec` has a feature called [url chaining](https://filesystem-spec.readthedocs.io/en/latest/features.html?highlight=simplecache#url-chaining), that's what makes the `::` do. Using url chaining, one can build layers of filesystems and do interesting tricks (like opening files within zipfiles stored on a remote http server).

`simplecache` is [a caching filesystem layer](https://filesystem-spec.readthedocs.io/en/latest/features.html?highlight=simplecache#caching-files-locally) which doesn't change the presentation of the contents, but keeps a copy on local disk (by default in a temporary folder which is deleted after the Python process stops). `simplecache` assumes that the content behind the referenced resource doesn't change which makes cache invalidation trivial.

So basically what the line does is: when requesting the file once within a running process, download the entire thing and keep for reuse it as long as the process is running. The change thus increases the upfront amount of data to be loaded but decreases the number of requests (it's not one to many per variable). For small files, this approach might be better in general than opendap.